### ISSUE TITLE
[unbound-method] False positive when comparing to null

### ISSUE THREAD

**Issue by phaux** [typescript-eslint/typescript-eslint#877]
**Repro**

```JSON
{
"extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking"
  ]
}
```

```TS
interface Options {
  onBegin?(): void
}

function foo(options: Options) {
  if (options.onBegin != null) options.onBegin()
}
```

**Expected Result**

No errors.

**Actual Result**

First `options.onBegin` is marked as error.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0` |
| `@typescript-eslint/parser`        | `2.0.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |
